Sources/Sentry/SentryScope.m

```diff
@@ -71,7 +71,15 @@ - (instancetype)initWithMaxBreadcrumbs:(NSInteger)maxBreadcrumbs
     if (self = [super init]) {
         self.listeners = [NSMutableArray new];
         self.maxBreadcrumbs = maxBreadcrumbs;
-        [self clear];
+        self.breadcrumbArray = [NSMutableArray new];
+        self.userObject = nil;
+        self.tagDictionary = [NSMutableDictionary new];
+        self.extraDictionary = [NSMutableDictionary new];
+        self.contextDictionary = [NSMutableDictionary new];
+        self.distString = nil;
+        self.environmentString = nil;
+        self.levelEnum = kSentryLevelNone;
+        self.fingerprintArray = [NSMutableArray new];
```

Comment:
L: I'd extract this to a private method like `clearFields` or something like that, so both methods `initWithMaxBreadcrumbs` and `clear` would call it, so we don't duplicate the lines nor need to test the same thing in 2 different methods.

Reply:
I strongly agree with that 👍